### PR TITLE
Tool should be uploaded to save-demo asynchronously

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Changed `asyncEffect` to `doOnSuccess` in order to initialize file download after the response is received

### CODE REVIEW

**Review by nulls — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt [hunk 1]
```diff
@@ -100,16 +101,18 @@ class DemoManagerController(
                 }
                 .toBodilessEntity()
         }
-        .flatMap {
+        .map {
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:104]: also suggestion to collect all required info to a single request and send it to demo (in `doOnSuccess` section) and here return a single `accepted`

**Review by nulls — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt (hunk 1 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:104]: the same as demo can fail when file started to upload.
but got you point and actually, Demo can return 202 on its side.
As I can see -- backend is just a proxy here (to check authorization and authentication).
So common suggestion to reduce logic on backend and move it to demo

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:104]: ok, may be done in next PR

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt [hunk 2]
```diff
@@ -100,16 +101,18 @@ class DemoManagerController(
...
+            StringResponse.ok("Successfully signed up ${demoCreationRequest.demoDto.projectCoordinates} demo.")
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt:105]: Done

**Pull request merged**